### PR TITLE
feat(providers): add connect_builtin example

### PR DESCRIPTION
**Pull request by yash-atreya**
- Updates the dependencies in `examples-providers`.
- Adds the `connect_builtin` examples for all three builtin transport type. HTTP, WS and IPC.

### CODE REVIEW

**Comment by yash-atreya** [comment]
This would fail due to the IPC path not being present.

**Comment by mattsse** [comment]
does this work locally?

it's possible that the file is created somewhere else, I think we want this as a getter on the anvil instance as well

**Comment by prestwich** [comment]
>  think we want this as a getter on the anvil instance as well

prefer the opposite. `ProviderBuilder::on_anvil(self, AnvilSettings)`

reasoning: node-bindings is nearly dep-free. we want provider to depend on it, not vice versa

**Comment by yash-atreya** [comment]
> does this work locally?
> 
> it's possible that the file is created somewhere else, I think we want this as a getter on the anvil instance as well

It works if the file exists but would ultimately fail with ```Socket operation on non-socket (os error 38)```. Because after all its a no

**Comment by zerosnacks** [comment]
> This would CI fail due to the IPC path not being present.

If you have an example that cannot be ran (see #31) you can update the filter applied in the CI here: https://github.com/alloy-rs/examples/blob/60d14dff23d8754a84a32ed4efdf36a2b1992cfb/.github/workflows/test.yml#L44-L55 and here: https://github.com/alloy-rs/examples/blob/60d14dff23d8754a84a32ed4efdf36a2b1992cfb/.github/workflows/integration.yml#L25-L37

File: examples/providers/examples/connect_builtin.rs
Comment (zerosnacks) [inline examples/providers/examples/connect_builtin.rs:19]: More of a general comment and a personal opinion but I'm not a fan of the `connect_builtin` name. I don't think it is particularly clear to users and sounds like an internal method rather than the equivalent of a `default`.

Comment (zerosnacks) [reply examples/providers/examples/connect_builtin.rs:19]: Possibly beyond the scope of the intended PR but I would also prefer to see the parameter being strongly typed (`Url`-like) instead of a `&str`

File: examples/providers/examples/connect_builtin.rs
Comment (zerosnacks) [inline examples/providers/examples/connect_builtin.rs:7]: Prefer to use `alloy` namespace

File: examples/providers/Cargo.toml
Comment (zerosnacks) [inline examples/providers/Cargo.toml:30]: Why are these required and not accessible on the `alloy` namespace?

Comment (yash-atreya) [reply examples/providers/Cargo.toml:30]: Compiler doesn't the recognize them as the same type when used with `alloy-provider` (w/o the namespace).

Comment (zerosnacks) [reply examples/providers/Cargo.toml]: Ah I see, then this should be resolved once that upstream PR has been merged

**Review by mattsse — APPROVED** [review]
lgtm, pending CI filter

**Review by zerosnacks — APPROVED** [review]

**Pull request merged**
